### PR TITLE
Add a section about coloring

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: _posts/2025-04-28-sparse-autodiff.md [hunk 1]
```diff
@@ -488,8 +489,59 @@ Instead, it returns an empty set.
...
 
+<!-- Coloring techniques? -->
 ### Matrix coloring
 
+Coloring techniques, such as row and column colorings, are used to determine the seed vectors needed to compute the minimal number of directional derivatives required to recover all the non-zero entries of Jacobians.
+A seed vector is a linear combination of the rows or columns of the identity matrix.
+When a directional derivative of a vector-valued function is taken along this seed vector, the result is a linear combination of the corresponding rows or columns of the Jacobian.
+
+Theses methods relies on reverse or forward automatic differentiation (AD) to efficiently recover the compressed rows or columns of the Jacobian.
+This is particularly valuable when dealing with sparse Jacobians, as it enables significant compression without overlap among the non-zero entries.
```
Comment (adrhill) [inline _posts/2025-04-28-sparse-autodiff.md:500]: Maybe this concept of a seed vector could be moved into the "Leveraging structure" section?

I feel like there is some overlap here.

**Review by adrhill — COMMENTED** [review]
Thanks Alexis!

**Comment by gdalle** [comment]
I made the coloring section more specifically about coloring, and also adjusted a few things around pattern detection. @adrhill I think this is good to go

**Comment by amontoison** [comment]
Why did you removed my figures? :cry:

**Comment by amontoison** [comment]
Well, you just removed all my text and figures...
I don't know why I worked on that if you remove all my contributions.

**Comment by gdalle** [comment]
As for the illustrations, it might be better to save realistic matrices for the last section, where we will demonstrate the Julia pipeline. In the theoretical stuff beforehand, we can probably keep using the same kind of simple images that Adrian has been creating. For instance, we could have one showing various column colorings on a very small matrix, with different numbers of colors in total.

**Comment by amontoison** [comment]
Maybe just open a branch and create a PR instead of hijacking mine?
You just removed all my modifications, Guillaume.
The repeating part was just about the seed, not the coloring and the figures. Am I wrong?

For the bicoloring, I’m not talking about our research and how we can approach it (relation with star and acyclic symmetric coloring); I’m just suggesting a nice comment that both AD methods can be combined.
I thought that this fit well into the workflow of the section and can easily illustrated.

I’m just disappointed because I felt that you removed my contribution, and I don’t feel legitimate putting my name on the blog.
I also spent time generating new figures one evening.

**Comment by gdalle** [comment]
> Maybe just open a branch and create a PR instead of hijacking mine? You just removed all my modifications, Guillaume.

My apologies for that, you are right that I should have opened a new branch. The reason I did this is because Adrian had left a comment yesterday asking for changes, and I didn't know if you planned to take it into account or not, so I thought I would make the changes myself.

We only have one week to get this blog post ready to the point of submission, so yes I was a bit assertive to ensure that we don't get stuck. I should have asked you first when you would have time to work on it.

> The repeating part was just about the seed, not the coloring and the figures. Am I wrong?

I think you are wrong. Everything you wrote apart from the mention of bicoloring was strictly included in what was already written above.

> For the bicoloring, I’m not talking about our research and how we can approach it (relation with star and acyclic symmetric coloring); I’m just suggesting a nice comment that both AD methods can be combined.

For me, not mentioning bicoloring meant no visualizations, perhaps a pointer to an article but that's about it. 

> I also spent time generating new figures one evening.

I thought I recognized the figures you had already made for our bicoloring paper, which is why I removed them, sorry. Ideally there should be no overlap with our own research, because these blog posts are forbidden to contain self-promotion. Besides, we don't have to reuse the exact same matrix from the SIAM Review, we can go with something simpler. The goal of this section is to give intuition about what the coloring means, whereas we can put realistic matrices in the final section.

> I’m just disappointed because I felt that you removed my contribution, and I don’t feel legitimate putting my name on the blog.

I understand your disappointment. I want you to contribute to this blog post because you're an essential part of the ecosystem the three of us built together. But the paragraphs you wrote about coloring were redundant with the rest, so I took the liberty to remove them.

**Comment by gdalle** [comment]
I have reverted my hijacking and opened #14 to propose alternative changes. Since @adrhill is taking the lead on this blog post, I'll let him decide how to combine the PRs

**Comment by adrhill** [comment]
Yeah, I'll close this in favor of #14, since this is a duplicate of the "Leveraging structure" section.
Let's build from there!

**Pull request closed without merging**